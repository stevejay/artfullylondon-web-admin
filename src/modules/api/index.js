import * as eventService from './sagas/event-service'
import * as imageService from './sagas/image-service'
import * as monitorService from './sagas/monitor-service'
import * as searchService from './sagas/search-service'
import * as tagService from './sagas/tag-service'
export { eventService, imageService, monitorService, searchService, tagService }
