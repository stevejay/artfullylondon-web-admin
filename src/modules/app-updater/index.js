import AppUpdater from './components/app-updater'
import sagas from './sagas'
export { AppUpdater, sagas }
