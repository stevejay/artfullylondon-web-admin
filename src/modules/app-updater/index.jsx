import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'

import FadeTransition from '_src/components/transition/fade'
import Button from '_src/components/button'
import { selectors, appActions } from '_src/store'
import './index.scss'

export class AppUpdater extends React.Component {
  handleClick = () => {
    this.props.dispatch(appActions.updateApp())
  }
  render () {
    const { shouldUpdate } = this.props

    return (
      <FadeTransition in={shouldUpdate} timeout={200}>
        <section styleName='container'>
          <p styleName='message'>
            <strong>A new version of the app is available!</strong>&nbsp;
            Click the Update button to reload the page and update.
          </p>
          <Button onClick={this.handleClick} type='submit'>
            Update
          </Button>
        </section>
      </FadeTransition>
    )
  }
}

AppUpdater.propTypes = {
  shouldUpdate: PropTypes.bool.isRequired,
  dispatch: PropTypes.func.isRequired
}

export default connect(state => ({
  shouldUpdate: selectors.shouldUpdate(state)
}))(AppUpdater)
