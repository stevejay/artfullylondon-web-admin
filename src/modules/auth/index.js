// @flow

import AuthHandler from "./components/auth-handler";
import LogoutMenuOptionHandler from "./components/logout-menu-option-handler";
import * as amplify from "./amplify";
import resolvers from "./graphql/resolvers";
const getAuthToken = amplify.auth.getAuthToken;
export { AuthHandler, LogoutMenuOptionHandler, getAuthToken, resolvers };
