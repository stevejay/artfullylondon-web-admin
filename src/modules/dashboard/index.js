import DashboardPage from '_src/modules/dashboard/pages/dashboard'
import { moduleName, reducer, selectors } from '_src/modules/dashboard/reducers'
import sagas from '_src/modules/dashboard/sagas'
import * as actions from '_src/modules/dashboard/actions'

export { DashboardPage, moduleName, reducer, selectors, sagas, actions }
