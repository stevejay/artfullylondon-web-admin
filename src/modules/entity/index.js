import EntityPage from './pages/entity'

import EntityFeaturedDetail from './components/featured-detail'
import EntityDescription from './components/description'
import EntitySectionHeading from './components/section-heading'
import EntityDetailsContainer from './components/details-container'
import EntityHeading from './components/heading'
import EntityWeSay from './components/we-say'
import EntityInfoBar from './components/info-bar'
import EntityColumnLayout from './components/column-layout'
import EntityColumn from './components/column'
import EntityAddress from './components/address'
import EntityDisabilityAccess from './components/disability-access'
import OpeningTimes from './components/opening-times'

import { moduleName, reducer, selectors } from './reducers'
import sagas from './sagas'
import * as actions from './actions'
import * as entityMapper from './lib/mapper'

export {
  EntityPage,
  EntityFeaturedDetail,
  EntityDescription,
  EntitySectionHeading,
  EntityDetailsContainer,
  EntityHeading,
  EntityWeSay,
  EntityInfoBar,
  EntityColumnLayout,
  EntityColumn,
  EntityAddress,
  EntityDisabilityAccess,
  OpeningTimes,
  entityMapper,
  moduleName,
  reducer,
  selectors,
  sagas,
  actions
}
