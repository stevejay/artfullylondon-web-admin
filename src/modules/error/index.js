import NotFoundPage from '_src/modules/error/pages/not-found'
export { NotFoundPage }
