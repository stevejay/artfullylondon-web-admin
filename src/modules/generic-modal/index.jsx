import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'
import Modal from '_src/components/modal'
import { hideModal } from '_src/actions/modal'

export const GenericModal = ({
  showingModal,
  modalComponent,
  hideModal,
  modalProps,
  modalComponentProps
}) => (
  <Modal
    show={showingModal}
    onHide={hideModal}
    modalProps={modalProps}
    component={modalComponent}
    componentProps={modalComponentProps}
  />
)

GenericModal.propTypes = {
  showingModal: PropTypes.bool.isRequired,
  modalProps: PropTypes.object,
  modalComponent: PropTypes.func,
  modalComponentProps: PropTypes.object,
  hideModal: PropTypes.func.isRequired
}

export default connect(
  state => ({
    showingModal: state.modal.showModal,
    modalProps: state.modal.modalProps,
    modalComponent: state.modal.component,
    modalComponentProps: state.modal.componentProps
  }),
  dispatch => ({
    hideModal: bindActionCreators(hideModal, dispatch)
  })
)(GenericModal)
