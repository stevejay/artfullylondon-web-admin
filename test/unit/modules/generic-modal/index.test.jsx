import React from 'react'
import { shallow } from 'enzyme'
import _ from 'lodash'

import { GenericModal } from '_src/modules/generic-modal'

it('should render correctly', () => {
  const wrapper = shallow(
    <GenericModal
      showingModal
      modalProps={{}}
      modalComponent={() => <div />}
      modalComponentProps={{}}
      hideModal={_.noop}
    />
  )

  expect(wrapper).toMatchSnapshot()
})
