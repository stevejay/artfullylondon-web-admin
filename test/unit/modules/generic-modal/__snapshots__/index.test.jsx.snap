// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<Modal
  component={[Function]}
  componentProps={Object {}}
  modalProps={Object {}}
  onHide={[Function]}
  show={true}
/>
`;
